optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -0,0 +1,623 @@
+        past_key_values (`tuple(tuple(torch.FloatTensor), *optional*)`
+            Contains the precomputed key and value hidden states of the attention blocks used to speed up decoding.
+            The tuple is of length `config.n_layers` with each tuple having 2 tensors of shape
+            `(batch_size, num_heads, decoder_sequence_length, embed_size_per_head)` and 2 additional tensors of shape
+            `(batch_size, num_heads, encoder_sequence_length, embed_size_per_head)`.
+"""
+
+_TOKENIZER_FOR_DOC = "AutoTokenizer"
+
+TRANSLATION_EXAMPLE = r"""
+    Example of text generation:
+
+    ```python
+    >>> from transformers import {processor_class}
+    >>> from optimum.onnxruntime import {model_class}
+
+    >>> tokenizer = {processor_class}.from_pretrained("{checkpoint}")
+    >>> model = {model_class}.from_pretrained("{checkpoint}")
+
+    >>> inputs = tokenizer("My name is Eustache and I like to", return_tensors="pt")
+
+    >>> gen_tokens = model.generate(**inputs)
+    >>> outputs = tokenizer.batch_decode(gen_tokens)
+    ```
+
+    Example using `transformers.pipeline`:
+
+    ```python
+    >>> from transformers import {processor_class}, pipeline
+    >>> from optimum.onnxruntime import {model_class}
+
+    >>> tokenizer = {processor_class}.from_pretrained("{checkpoint}")
+    >>> model = {model_class}.from_pretrained("{checkpoint}")
+    >>> onnx_summarization = pipeline("translation_en_to_de", model=model, tokenizer=tokenizer)
+
+    >>> text = "My name is Eustache."
+    >>> pred = onnx_summarization(text)
+    ```
+"""
+
+
+@add_start_docstrings(
+    """
+    Sequence-to-sequence model with a language modeling head for ONNX Runtime inference.
+    """,
+    ONNX_INPUTS_DOCSTRING,
+)
+class ORTModelForConditionalGeneration(ORTModel):
+    # Used in from_transformers to export model to onnx
+    pipeline_task = "seq2seq-lm"
+    auto_model_class = AutoModelForSeq2SeqLM
+
+    def __init__(
+        self,
+        encoder_session: onnxruntime.InferenceSession = None,
+        decoder_session: onnxruntime.InferenceSession = None,
+        decoder_with_past_session: onnxruntime.InferenceSession = None,
+        config: transformers.PretrainedConfig = None,
+        **kwargs
+    ):
+        self.config = config
+        self.model_save_dir = kwargs.get("model_save_dir", None)
+        self._device = get_device_for_provider(encoder_session.get_providers()[0])
+        self.encoder = ORTEncoder(session=encoder_session, device=self._device)
+        self.decoder = ORTDecoder(session=decoder_session, device=self._device)
+        self.decoder_with_past = ORTDecoder(session=decoder_with_past_session, device=self._device)
+        self.encoder_file_name = kwargs.get("encoder", ONNX_ENCODER_NAME)
+        self.decoder_file_name = kwargs.get("decoder", ONNX_DECODER_NAME)
+        self.decoder_file_with_past_name = kwargs.get("decoder_with_past", ONNX_DECODER_WITH_PAST_NAME)
```

Comment:
Does a user always want to provide a decoder with past keys? I know the french-guy was doing some investigation that past_keys do not always provide better performance. 
Should me make it optional? that you only need encoder + decoder? 

have you run any "benchmarks" to validate this?

Reply:
I have seen some issues for case where inference was [slower](https://github.com/NVIDIA/TensorRT/issues/1807) without pre-computed past key/values but never did a benchmark on my own. In case where generation is done on a small model or with short sequence length I imagine it could be interesting for the user to run inference without the past key/values. I can add this feature in an other PR but suggest we merge this #199 first.

Reply:
Will add this in #241